src/ui/testrtc-test.html

```diff
@@ -137,8 +138,18 @@
       done: function() {
         this.setProgress(null);
         var success =
-            (this.errorCount === 0 && (this.successCount + this.warningCount) > 0);
-        this.state = (success ? 'success' : (this.isDisabled ? 'disabled' : 'failure'));
+          (this.errorCount + this.warningCount === 0 && this.successCount > 0);
+
+        if (success) {
+          this.state = 'success';
+        } else if (this.errorCount > 0) {
+          this.state = 'failure';
+        } else if (this.warningCount > 0) {
```

Comment:
Yes thank you for pointing this out, I thought about exactly this but then I forgot to do anything about it.